segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/AutoScaleMonitor.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.host.stat;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.pravega.client.EventStreamClientFactory;
+import io.pravega.common.concurrent.ExecutorServiceHelpers;
+import io.pravega.segmentstore.contracts.StreamSegmentStore;
+import java.util.concurrent.ScheduledExecutorService;
+import lombok.Data;
+import lombok.Getter;
+import lombok.NonNull;
+
+/**
+ * Process-wide monitor for auto-scaling events. Attaches to a {@link StreamSegmentStore} and provides a {@link SegmentStatsRecorder}
+ * that can be used to record segment-related statistics. All scale events are reported using a {@link EventStreamClientFactory}.
+ */
+@Data
+public class AutoScaleMonitor implements AutoCloseable {
+    private final ScheduledExecutorService executor = ExecutorServiceHelpers.newScheduledThreadPool(10, "auto-scaler");
```

Comment:
Do we need a core size of 10 threads in this pool or could we do with less?

Reply:
This is how it was originally. The subsequent PR for this will make this configurable. I know there's a lot of promises here, but I decided to split the work in two phases to make things clearer.